services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/docker_logs.py

```diff
@@ -0,0 +1,108 @@
+import logging
+from asyncio import CancelledError, Task, create_task
+from contextlib import suppress
+from typing import Any, Callable, Coroutine, Dict, Optional, cast
+
+from fastapi import FastAPI
+
+from .rabbitmq import RabbitMQ
+from .utils import docker_client
+
+logger = logging.getLogger(__name__)
+
+
+async def _logs_fetcher_worker(
+    container_name: str, dispatch_log: Callable[..., Coroutine[Any, Any, None]]
+) -> None:
+    logger.info("Started log fetching for container %s", container_name)
+    async with docker_client() as docker:
+        container = await docker.containers.get(container_name)
+
+        # extact image to display in logs, Eg: from
+        # registry:5000/simcore/services/dynamic/dy-static-file-server-dynamic-sidecar:2.0.2
+        # "dy-static-file-server-dynamic-sidecar:2.0.2"
+        container_inspect = await container.show()
+        image_name = container_inspect["Config"]["Image"].split("/")[-1]
+
+        logger.debug("Streaming logs from %s, image %s", container_name, image_name)
+        async for line in container.log(stdout=True, stderr=True, follow=True):
+            await dispatch_log(image_name=image_name, message=line)
+
+
+class BackgroundLogFetcher:
+    def __init__(self, app: FastAPI) -> None:
+        # requires rabbitmq to be in place
+        assert app.state.rabbitmq  # nosec
+
+        self._app: FastAPI = app
+
+        self._log_processor_tasks: Dict[str, Task[None]] = {}
+
+    @property
+    def rabbitmq(self) -> RabbitMQ:
+        return self._app.state.rabbitmq  # type: ignore
+
+    async def _dispatch_logs(self, image_name: str, message: str) -> None:
+        # sending the logs to the UI to facilitate the
+        # user debugging process
+        await self.rabbitmq.post_log_message(f"[{image_name}] {message}")
+
+    async def start_log_feching(self, container_name: str) -> None:
+        self._log_processor_tasks[container_name] = create_task(
+            _logs_fetcher_worker(
+                container_name=container_name, dispatch_log=self._dispatch_logs
+            )
+        )
+
+        logger.info("Subscribed to fetch logs from '%s'", container_name)
+
+    async def stop_log_fetching(self, container_name: str) -> None:
+        logger.debug("Stopping logs fetching from container '%s'", container_name)
+        task = self._log_processor_tasks[container_name]
+        with suppress(CancelledError):
+            task.cancel()
```

Comment:
minor: the call to cancel does not need to be in the context manager. the one throwing is the ```await task```